### PR TITLE
ISU: support real-world insitu datasets

### PR DESCRIPTION
**Pull request by ChaoTUDelft**
## Summary
- Full ISU → QCL → DPR → SDI integration test added using two real piezometer datasets
- ISU parser now handles GBK/GB18030-encoded CSV files and multi-bit QC flag strings
- Detection scoring extended to recognise DataSet-series columns and (mm) unit columns
- Pipeline bug fixes in DPR (ZIP packaging) and SDI (STAC column type, QC result interface)

## Context

Two real-world piezometer datasets were used to validate the full ISU pipeline:
- **Piezometer1.csv** – UTF-8, multi-depth deployment, QC column as 18-bit flag strings (`000000000000000000`)
- **Piezometer2.csv** – GBK-encoded, dam monitoring, DataSetI–IV columns + X/Y/H(mm) displacements

## Changes

- `parsers/base.py`: multi-encoding CSV reader (UTF-8/GBK/GB18030/Latin-1 fallback); fix `ensure_qc_column()` to normalise all-zero bit-flag strings (e.g. `000000000000000000`) to `1` (pass)
- `parsers/slope.py`: extend detection keywords (`dataset`, `celsius`, `kilopascal`, `pascal`, `(mm)` units); two-pass CSV read to preserve QC column as string before pandas type inference
- `pipeline.py`: add `dataset`, `kilopascal`, `pascal` to piezometer sensor type keywords
- `dpr/data_export`: fix ZIP packaging to store files by basename (absolute path caused FileNotFoundError in SDI)
- `sdi/loader`: fix `col.get('type', 'TEXT')` fallback (KeyError on STAC assets without type field)
- `sdi/__init__`: add `store_qc_result()` stub for QCL dispatch contract 

## Tests

Added `TestISUIntegration.test_integration_ISU_002` (parametrized over both datasets):
- Full pipeline: ISU parse → QCL validate → DPR STAC → SDI package import
- PostGIS and STAC API calls are mocked; no infrastructure required
- Test assets: `tests/assets/isu/Piezometer1.csv`, `tests/assets/isu/Piezometer2.csv`

**TODO for SDI**: replace stub with real persistence logic (PostGIS table / STAC record).

### CODE REVIEW

File: subsystems/isu/etl_engine/parsers/base.py [hunk 1]
```diff
@@ -1,9 +1,24 @@
 from abc import ABC, abstractmethod
-from typing import List, Dict, Any, Optional
+from typing import List, Dict, Any, Optional, Tuple
 import pandas as pd
 import io
 
 
+_ENCODINGS_TO_TRY = ['utf-8', 'gbk', 'gb18030', 'latin-1']
```
Comment (landam) [inline subsystems/isu/etl_engine/parsers/base.py]: To be move to `config.yaml`

**Review by landam — CHANGES_REQUESTED** [review]

File: subsystems/isu/etl_engine/parsers/base.py [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
+
+
+def _read_csv_bytes(content: bytes, **kwargs) -> Tuple[pd.DataFrame, str]:
```
Comment (janruzicka-cybele) [inline subsystems/isu/etl_engine/parsers/base.py]: Maybe move inside the class? I am not sure why it should not be incorporated into the BaseParser class.

File: subsystems/isu/etl_engine/parsers/slope.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import pandas as pd
 import io
 import os
-from .base import BaseParser
+from .base import BaseParser, _read_csv_bytes
```
Comment (janruzicka-cybele) [inline subsystems/isu/etl_engine/parsers/slope.py]: If the _read_csv_bytes is part of the BaseParser the code will be cleaner.

File: tests/test_integration.py [hunk 4]
```diff
@@ -140,3 +145,92 @@ def test_integration_ISU_001(self, tmp_path):
 
         importer = InSituDataLoader(zip_path=exported_temp.name)
         importer.import_zip()
+
+
+# Dummy pg/stac config for mocked SDI infrastructure calls
+_SDI_PG_CONFIG = {
```
Comment (janruzicka-cybele) [inline tests/test_integration.py]: Not sure why is not used the real STAC that is part of the environment.

**Review by janruzicka-cybele — CHANGES_REQUESTED** [review]

**Review by janruzicka-cybele — APPROVED** [review]

**Review by landam — APPROVED** [review]
